ga4gh/backend.py

```diff
@@ -700,6 +710,38 @@ def featuresGenerator(self, request):
             request.page_token, request.page_size,
             request.feature_types, parentId, request.name, request.gene_symbol)
 
+    def phenotypesGenerator(self, request):
+        """
+        Returns a generator over the (phenotypes, nextPageToken) pairs
+        defined by the (JSON string) request
+        """
+        # TODO make paging work using SPARQL?
```

Comment:
At some point in the past, we had looked at implementing paging within the datamodel.  At the time, it seemed that for; conformance with the implementation style with other runners in backend, considering the size of the data (tens, hundred of evidence items not millions) and the discussions about streaming, that we left it as is.

The following tests address paging.  
https://github.com/ga4gh/server/blob/g2p/tests/end_to_end/test_g2p.py#L468
https://github.com/ga4gh/server/blob/g2p/tests/end_to_end/test_g2p.py#L482
https://github.com/ga4gh/server/blob/g2p/tests/end_to_end/test_g2p.py#L495

Suggest considering re-implementation as a performance improvement when needed